WooCommerce/Classes/Authentication/Store Creation/Profiler/StoreCreationProfilerQuestionContainerViewModel.swift

```diff
@@ -36,22 +36,28 @@ final class StoreCreationProfilerQuestionContainerViewModel: ObservableObject {
 
     init(storeName: String,
          analytics: Analytics = ServiceLocator.analytics,
-         onCompletion: @escaping (SiteProfilerData?) -> Void) {
+         onCompletion: @escaping (StoreProfilerAnswers?) -> Void) {
         self.storeName = storeName
         self.analytics = analytics
         self.completionHandler = onCompletion
     }
 
+    func onAppear() {
+        analytics.track(event: .StoreCreation.siteCreationStep(step: .profilerSellingStatusQuestion))
+    }
+
     func saveSellingStatus(_ answer: StoreCreationSellingStatusAnswer?) {
         if answer == nil {
             analytics.track(event: .StoreCreation.siteCreationProfilerQuestionSkipped(step: .profilerSellingStatusQuestion))
         } else if let answer,
                     answer.sellingStatus == .alreadySellingOnline,
-                    answer.sellingPlatforms == nil {
+                  answer.sellingPlatforms == nil || answer.sellingPlatforms?.isEmpty == true {
             analytics.track(event: .StoreCreation.siteCreationProfilerQuestionSkipped(step: .profilerSellingPlatformsQuestion))
         }
+
         sellingStatus = answer
         currentQuestion = .category
+        analytics.track(event: .StoreCreation.siteCreationStep(step: .profilerCategoryQuestion))
```

Comment:
❓ I see that we're tracking the step only if the user moves forward. Do we want to track the step when the tap back too? If so, we might need to observe the `currentQuestion` value to track the step instead.

Reply:
Good point. The previous implementation doesn't track an event when the user navigates back. I think we can stick to that behaviour. Let me know if you think we should change that.

Reply:
I think it depends on the definition of the event, whether it's strictly about when the step is displayed or when the step is first shown. Either way, if not tracking the back navigation doesn't cause any issue with Tracks funnels, we can keep the behavior as-is.